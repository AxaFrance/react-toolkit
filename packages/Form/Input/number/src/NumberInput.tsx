import React, {
  ComponentPropsWithoutRef,
  ComponentPropsWithRef,
  ReactNode,
} from 'react';

import {
  Field,
  HelpMessage,
  FieldInput,
  useInputClassModifier,
} from '@axa-fr/react-toolkit-form-core';
import { useId } from '@axa-fr/react-toolkit-core';
import Number from './Number';

type Props = ComponentPropsWithoutRef<typeof Field> &
  ComponentPropsWithRef<typeof Number> & {
    helpMessage?: ReactNode;
    children?: ReactNode;
  };

const NumberInput = ({
  classModifier,
  message,
  children,
  helpMessage,
  id,
  label,
  classNameContainerLabel,
  classNameContainerInput,
  forceDisplayMessage,
  messageType,
  isVisible,
  className,
  disabled,
  ...otherProps
}: Props) => {
  const inputId = useId(id);
  const { inputClassModifier, inputFieldClassModifier } = useInputClassModifier(
    classModifier,
    disabled,
    !!children
  );
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        className="af-form__text"
        classModifier={inputFieldClassModifier}>
        <Number
          id={inputId}
          classModifier={inputClassModifier}
          disabled={disabled}
          {...otherProps}
        />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

export default NumberInput;
