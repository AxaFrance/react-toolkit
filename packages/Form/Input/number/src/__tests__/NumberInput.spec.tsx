import React from 'react';
import { render } from '@testing-library/react';
import NumberInput from '../NumberInput';

describe('<NumberInput>', () => {
  it('renders NumberInput correctly', () => {
    const { asFragment } = render(
      <NumberInput
        label="Image *"
        id="iddateinput"
        name="placeImage"
        value={2}
        onChange={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
