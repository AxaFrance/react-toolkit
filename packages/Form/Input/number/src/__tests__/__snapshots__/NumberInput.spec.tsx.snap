// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput> renders NumberInput correctly 1`] = `
<DocumentFragment>
  <div
    class="row af-form__group"
  >
    <div
      class="col-md-2"
    >
      <label
        class="af-form__group-label"
        for="iddateinput"
      >
        Image *
      </label>
    </div>
    <div
      class="col-md-10"
    >
      <div
        class="af-form__text"
      >
        <input
          class="af-form__input-number"
          id="iddateinput"
          name="placeImage"
          type="number"
          value="2"
        />
      </div>
      <small
        class="af-form__help"
      />
    </div>
  </div>
</DocumentFragment>
`;
